interpreter/cling/lib/Interpreter/NullDerefProtectionTransformer.cpp

```diff
@@ -101,100 +50,37 @@ namespace cling {
   public:
     IfStmtInjector(Sema& S) : m_Sema(S), m_Context(S.getASTContext()),
     m_LookupResult(0) {}
-    CompoundStmt* Inject(CompoundStmt* CS) {
-      NodeContext result = VisitCompoundStmt(CS);
-      return cast<CompoundStmt>(result.getStmt());
+    void Inject(CompoundStmt* CS) {
+      VisitStmt(CS);
     }
 
-    NodeContext VisitStmt(Stmt* S) {
-      return NodeContext(S);
-    }
-
-    NodeContext VisitCompoundStmt(CompoundStmt* CS) {
-      ASTContext& C = m_Sema.getASTContext();
-      llvm::SmallVector<Stmt*, 16> stmts;
-      for (CompoundStmt::body_iterator I = CS->body_begin(), E = CS->body_end();
-           I != E; ++I) {
-        NodeContext nc = Visit(*I);
-        if (nc.isSingleStmt())
-          stmts.push_back(nc.getStmt());
-        else
-          stmts.append(nc.getStmts().begin(), nc.getStmts().end());
+    void VisitStmt(Stmt* S) {
+      for (auto child: S->children()) {
+        if (child)
+          Visit(child);
       }
-
-      llvm::ArrayRef<Stmt*> stmtsRef(stmts.data(), stmts.size());
-      CompoundStmt* newCS = new (C) CompoundStmt(C, stmtsRef,
-                                                 CS->getLBracLoc(),
-                                                 CS->getRBracLoc());
-      return NodeContext(newCS);
-    }
-
-    NodeContext VisitIfStmt(IfStmt* If) {
-      // check the condition
-      NodeContext cond = Visit(If->getCond());
-      if (!cond.isSingleStmt())
-        If->setCond((clang::Expr*)cond.getStmt());
-
-      // No default constructor..
-      NodeContext result(If);
-      return result;
     }
 
-    NodeContext VisitCastExpr(CastExpr* CE) {
-      NodeContext castExpr = Visit(CE->getSubExpr());
-      if (castExpr.isSingleStmt())
-        CE->setSubExpr((clang::Expr*)castExpr.getStmt());
-
-      // No default constructor..
-      NodeContext result(CE);
-      return result;
-    }
-
-    NodeContext VisitUnaryOperator(UnaryOperator* UnOp) {
+    void VisitUnaryOperator(UnaryOperator* UnOp) {
+      Visit(UnOp->getSubExpr());
       if (UnOp->getOpcode() == UO_Deref) {
-        UnOp->setSubExpr((clang::Expr*)SynthesizeCheck(UnOp->getLocStart(),
-                                 UnOp->getSubExpr()));
-      }
-
-      // No default constructor..
-      NodeContext result(UnOp);
-      return result;
-    }
-
-    NodeContext VisitBinaryOperator(BinaryOperator* BinOp) {
-      // Here we might get if(check) throw; binop rhs.
-      NodeContext rhs = Visit(BinOp->getRHS());
-      // Here we might get if(check) throw; binop lhs.
-      NodeContext lhs = Visit(BinOp->getLHS());
-
-      if (rhs.isSingleStmt()) {
-        // FIXME:we need to loop from 0 to n-1
-        BinOp->setRHS((clang::Expr*)rhs.getStmt());
+        Expr* newSubExpr = SynthesizeCheck(UnOp->getLocStart(),
+                                 UnOp->getSubExpr());
```

Comment:
Indentation seems to be off? Can you align the `UnOp->`s (i.e. func args should be aligned across lines, unless the line is longer than 80 chars)